pagopa/io-app
# chore: [IOBP-611] Fix navigation on payment success

> [!WARNING]
> This PR depends on https://github.com/pagopa/io-app/pull/5656

## Short description
This PR fixes the navigation on payment success, which allows to navigate to the home screen once a payment is done, ore return back where the payment started (eg.: message details screen).

## List of changes proposed in this pull request
- Add the correct screen navigation to the Payments landing screen CTA
- Removed `startRoute` from `initPaymentState` action payload and `payments` redux store, which is no more required
- Added `onSuccess` prop to `initPaymentState` payload, which tells the payment flow where to navigate after a payment is completed ( show transaction, show home, return to the starting route)
- Refactored some navigation actions to allow a correct back navigation with `pop` or `popToTop`.

## How to test

> [!IMPORTANT]
> The messages feature does not support yet the new wallet section. It needs a bit of refactoring which I do not have enough context to do. To test this PR you need to replace `initializeAndNavigateToWalletForPayment` function and use the `usePagoPaPayment` hook where needed.

With the `io-dev-api-server,` enable the **New wallet section** FF from the **Profile** screen.
Start a new payment flow from the Payments landing screen. On a successfull payment, tap on Close button and check that the app navigates to the home screen.

## Preview

With `showHome` action
 <video src="https://github.com/pagopa/io-app/assets/6160324/86f13d26-3586-4566-9244-1cbb6c424e16" />

With `undefined` action (not provided) (return to starting screen)

<video src="https://github.com/pagopa/io-app/assets/6160324/9aa72fcc-a9ed-4b3a-9d4c-9f6e18d375bf" />
